test/system/test_api_interfaces.py

```diff
@@ -91,10 +91,62 @@ def test_delete_and_return_true(self):
             self.assertNotIn('Loopback0', config[0]['interfaces'])
 
     def test_delete_ethernet_raises_not_implemented_error(self):
-        with self.assertRaises(NotImplementedError):
-            for dut in self.duts:
+        for dut in self.duts:
+            with self.assertRaises(NotImplementedError):
                 dut.api('interfaces').delete(random_interface(dut))
 
+    def test_create_and_delete_ethernet_sub_interface(self):
```

Comment:
I meant the set_encap method.
Eg
1. Default Et1
2. Create Et1.1
3. Set_encap(et1.1). 

Number 3 should raise an error since the parent intf isn't routed, right? I haven't tested.